### PR TITLE
Improvement/rpn1 61 migrate code to java common

### PR DESCRIPTION
**Pull request by Sajjon**

### CODE REVIEW

File: src/main/java/com/radixdlt/DefaultSerialization.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.radixdlt.serialization.core.ClasspathScanningSerializationPolicy;
+import com.radixdlt.serialization.core.ClasspathScanningSerializerIds;
+
+public final class DefaultSerialization {
+
+	private DefaultSerialization() {
+		//not called
+	}
+
+	private static final Serialization defaultInstance = Serialization.create(ClasspathScanningSerializerIds.create(),
```
Comment (flotothemoon) [inline src/main/java/com/radixdlt/DefaultSerialization.java]: Why is this necessary?

**Review by flotothemoon — CHANGES_REQUESTED** [review]
This is fine on its own, declining to make that the other related PRs are all in line

File: src/main/java/com/radixdlt/DefaultSerialization.java (hunk 1 above)
Comment (clstrfsck) [reply src/main/java/com/radixdlt/DefaultSerialization.java]: It used to be in serialization, but is specific to radix-engine-library.  radixdlt-java uses a different class search mechanism.

File: build.gradle [hunk 2]
```diff
@@ -67,28 +67,9 @@ jacocoTestReport {
 }
 
 dependencies {
-    compile 'org.slf4j:slf4j-api:1.7.26'
-    compile 'org.slf4j:slf4j-simple:1.7.26'
 
-    compile('org.reflections:reflections:0.9.11') {
-        exclude group: 'com.google.guava', module: 'guava'
-    }
+    api project(':radixdlt-java-common')
```
Comment (clstrfsck) [inline build.gradle]: Should really be using the jitpack artifact here, something like:
```
    api 'com.github.radixdlt:radixdlt-java-common:rc~1.0-beta.1-SNAPSHOT'
```

File: examples/build.gradle [hunk 3]
```diff
@@ -34,4 +34,10 @@ dependencies {
     compile rootProject //TODO: Hack for now since in code freeze
     testCompile group: 'junit', name: 'junit', version: '4.12'
     testCompile 'org.hamcrest:hamcrest-library:1.3'
+
+    implementation('com.radixdlt:radixdlt-java-common') {
```
Comment (clstrfsck) [inline examples/build.gradle]: Shouldn't need this here, as it is API in ../build.gradle.

File: settings.gradle [hunk 4]
```diff
@@ -1,2 +1,5 @@
 rootProject.name = 'radix-engine-library'
 include 'examples'
+
+include ":radixdlt-java-common"
+project(":radixdlt-java-common").projectDir = file("../radixdlt-java-common")
```
Comment (clstrfsck) [inline settings.gradle]: See other comments on using jitpack artifact.

File: src/main/java/com/radixdlt/DefaultSerialization.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+			ClasspathScanningSerializationPolicy.create());
+
+	public static Serialization getDefault() {
+		return defaultInstance;
```
Comment (clstrfsck) [inline src/main/java/com/radixdlt/DefaultSerialization.java]: Would be better implemented using Holder initialisation pattern.

https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom

**Review by clstrfsck — CHANGES_REQUESTED** [review]
Would much prefer if we used the Jitpack dependencies, and used the holder pattern in DefaultSerialization, otherwise looks good.

File: examples/build.gradle (hunk 3 above)
Comment (Sajjon) [reply examples/build.gradle]: fixed

File: src/main/java/com/radixdlt/DefaultSerialization.java (hunk 5 above)
Comment (Sajjon) [reply src/main/java/com/radixdlt/DefaultSerialization.java]: fixed

File: build.gradle (hunk 2 above)
Comment (flotothemoon) [reply build.gradle]: Yes, same in the other PRs

File: src/main/java/com/radixdlt/DefaultSerialization.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ */
+
+package com.radixdlt;
+
+import com.radixdlt.serialization.Serialization;
+import com.radixdlt.serialization.core.ClasspathScanningSerializationPolicy;
+import com.radixdlt.serialization.core.ClasspathScanningSerializerIds;
+
+public final class DefaultSerialization {
```
Comment (flotothemoon) [inline src/main/java/com/radixdlt/DefaultSerialization.java:26]: Document me

Comment (Sajjon) [reply src/main/java/com/radixdlt/DefaultSerialization.java:26]: good catch!

**Comment by Sajjon** [comment]
Fixed gradle (with jitpack), ready for last review and merge :)

**Review by clstrfsck — APPROVED** [review]
Looks good to me.

**Review by talekhinezh — APPROVED** [review]
